langkit/examples/tutorials/Segmented_Validation_Example.ipynb

```diff
@@ -0,0 +1,277 @@
+          "text": [
+            "WARNING:whylogs.api.writer.whylabs:About to upload a profile with a dataset_timestamp that is in the future: -100.37037396430969s old.\n",
+            "WARNING:whylogs.api.writer.whylabs:About to upload a profile with a dataset_timestamp that is in the future: -98.07086396217346s old.\n",
+            "WARNING:whylogs.api.writer.whylabs:About to upload a profile with a dataset_timestamp that is in the future: -96.56739091873169s old.\n"
+          ]
+        }
+      ],
+      "source": [
+        "from typing import Any, Dict, Optional\n",
+        "import uuid\n",
+        "from random import randint\n",
+        "import os\n",
+        "\n",
+        "os.environ[\"TOKENIZERS_PARALLELISM\"] = \"false\"  # quiets a warning message\n",
+        "\n",
+        "from whylogs.core.metrics import FrequentItemsMetric\n",
+        "from whylogs.core.resolvers import MetricSpec\n",
+        "from whylogs.core.segmentation_partition import segment_on_column\n",
+        "from whylogs.core.validators import ConditionValidator\n",
+        "from whylogs.experimental.core.udf_schema import UdfSchema, udf_schema\n",
+        "\n",
+        "from langkit.openai import send_prompt\n",
+        "\n",
+        "# This pulls in some basic LangKit LLM metrics. We can pull in other/additional\n",
+        "# LangKit metrics, or register our own UDFs to include in the logged data.\n",
+        "from langkit import llm_metrics\n",
+        "\n",
+        "import whylogs as why\n",
+        "\n",
+        "\n",
+        "why.init()\n",
+        "\n",
+        "\n",
+        "# This runs when a prompt fails validation\n",
+        "def action_on_failure(validator_name: str, condition_name: str, value: Any, id: Optional[Any]=None):\n",
+        "  print(f\"Validator '{validator_name}' failed for condition '{condition_name}' on value '{value}' {'row id' if id else ''} {id if id else ''}\")\n",
+        "  # throw an exception here if you want. It will throw before the data is logged\n",
+        "\n",
+        "\n",
+        "validator = ConditionValidator(\n",
+        "    name = \"valid prompt\",\n",
+        "    conditions = {\"error code is OK\": lambda x: x == \"OK\"},\n",
+        "    actions = [action_on_failure]\n",
+        ")\n",
+        "\n",
+        "# apply any LangKit metrics (or other registered UDFs) and segment on the\n",
+        "# error code column. Also specifies the validator(s) to run on each column.\n",
+        "# Here we just validate the error_code column. Since we're segmenting on it,\n",
+        "# we add a frequent items metric to the error_code column too.\n",
+        "schema = udf_schema(\n",
+        "    segments=segment_on_column(\"error_code\"),\n",
+        "    validators={\"error_code\": [validator]}\n",
+        ")\n",
+        "schema.add_resolver_spec(\"error_code\", metrics=[MetricSpec(FrequentItemsMetric)])\n",
+        "\n",
+        "logger = why.logger(\n",
```

Comment:
The RollingLogger now accepts a schema parameter, so you could use that now.

Reply:
I'm not sure what you mean by this?